UnityProject/Assets/Scripts/Systems/CraftingV2/ResultHandlers/IResultHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Systems.CraftingV2.ResultHandlers
+{
+	/// <summary>
+	/// 	The class whose heirs can handle specified craft actions. For example, we can set a spear's damage
+	/// 	according to a glass shard used for crafting.
+	/// </summary>
+	[Serializable]
+	public class IResultHandler : ScriptableObject
+	{
```

Comment:
I might be blind but i cant find which Class uses this?

Reply:
`CraftingRecipe` uses this in the `CompleteCrafting` method:

![image](https://user-images.githubusercontent.com/42589308/128698392-fe2b4eed-2645-4215-9886-c496eb8c9d50.png)

Reply:
"The class whose heirs can handle specified craft actions." I cant find the class which inherits from IResultHandler

Reply:
Yeah, I haven't found a good use for the current list of recipes, which is very small atm, but I tested this method on a useless IResultHandler and it works fine

Reply:
This is necessary according to the bounty requirements:

> - Some requirements are also parts. Parts are either put into the result or otherwise alter its behavior.
>    - For example, what kind of bottle (or carton) you use to make a molotov cocktail affects its volume (and safety) and its held reagents are transferred over to the resulting molotov.
>    - For another example, whether you use a normal glass shard or plasma glass shard affects the resulting spear's damage and sprite.

Reply:
ah ok, thats fine.

Reply:
IResultHandler might be better as an interface, we can worry about that later if its hard for people to alter specific behaviours